app/src/main/java/com/example/topmovies/data/dao/MoviesDao.kt

```diff
@@ -0,0 +1,47 @@
+package com.example.topmovies.data.dao
+
+import androidx.room.*
+import com.example.topmovies.data.models.entity.MovieEntity
+import com.example.topmovies.data.repository.MoviesRepository.*
+import com.example.topmovies.domain.exeption.Failure
+import com.example.topmovies.domain.utils.ResultOf
+import com.example.topmovies.presentation.models.Movie
+
+@Dao
+abstract class MoviesDao {
```

Comment:
Is there any particular reason for appending "Entity" to each method?
Looks a bit redundant to me.